geode-core/src/main/java/org/apache/geode/distributed/internal/membership/InternalDistributedMember.java

```diff
@@ -641,4 +641,8 @@ public UUID getUUID() {
   public interface HostnameResolver {
     InetAddress getInetAddress(ServerLocation location) throws UnknownHostException;
   }
+
+  public MemberIdentifier getMemberIdentifier() {
+    return memberIdentifier;
+  }
```

Comment:
You don't need to expose the `MemberIdentifier` here, you can directly use `InternalDistributedMember.compareTo(DistributedMember o, boolean compareMemberData, boolean compareViewIds)`, which internally delegates to the `MemberIdentifier` class.